src/Internal/Processor.php

```diff
@@ -805,12 +806,15 @@ private function handleHandshake($packet)
     /** @see 14.6.4.1.2 LOCAL INFILE Request */
     private function handleLocalInfileRequest($packet)
     {
-        // @TODO async file fetch @rdlowrey
-        $file = \file_get_contents($packet);
-        if ($file !== "") {
-            $this->sendPacket($file);
-        }
-        $this->sendPacket("");
+        \Amp\asyncCall(function () use ($packet) {
+            $filePath = \substr($packet, 1);
+            /** @var \Amp\File\Handle $fileHandle */
+            $fileHandle = yield \Amp\File\open($filePath, 'r');
+            while (!$fileHandle->eof()) {
+                $this->sendPacket(yield $fileHandle->read());
```

Comment:
What if `read()` returns an empty chunk?

Reply:
During testing `eof` check worked fine. But just in case I addressed your concern in a new commit.